server/routes/subscription/createSubscription.ts

```diff
@@ -19,17 +20,71 @@ export default async (models, req: Request, res: Response, next: NextFunction) =
     return next(new Error(Errors.InvalidNotificationCategory));
   }
 
+  let obj;
+  const parsed_object_id = req.body.object_id.split(/-|_/);
+  const p_id = parsed_object_id[1];
+  const p_entity = parsed_object_id[0];
+  let chains, chainsIds;
+
+  switch (category.name) {
+    case 'new-thread-creation': {
+      chains = await models.Chain.findAll();
+      chainsIds = chains.map((c) => c.id);
+      if (chainsIds.includes(p_entity)) {
```

Comment:
Why not do a `findOne` rather than a `findAll` and search? Then we can also see if the entity even exists, rather than default to community (even if the community might not exist).

Reply:
good idea. I'll also add a query for the community too.